### PR TITLE
docs: Add .Net caching to docs

### PR DESCRIPTION
**Pull request by novakzaballa**
Thanks for submitting a PR! Please check the boxes below:

- [X] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [X] I have added information to `docs/` if required so people know about the feature!
- [X] I have filled in the "Changes" section below?
- [X] I have filled in the "How did you test this code" section below?
- [X] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes
Add content about .Net caching to docs

## How did you test this code?
Tested locally running docusaurus

### CODE REVIEW

File: docs/docs/clients/server-side.md [hunk 1]
```diff
@@ -1156,36 +1156,40 @@ _flagsmithClient = new FlagsmithClient(
     # Override the default Flagsmith API URL if you are self-hosting.
     # Optional.
     # Defaults to https://edge.api.flagsmith.com/api/v1/
-    apiUrl: "https://flagsmith.myproject.com"
+    apiUrl: "https://flagsmith.myproject.com",
+
+    # Enables caching support.
```
Comment (dabeeeenster) [inline docs/docs/clients/server-side.md]: This should be lower down in the config options - follow the other languages that have caching

File: docs/docs/clients/server-side.md [hunk 2]
```diff
@@ -1754,6 +1758,25 @@ cache.invalidate(projectLevelCacheKey);
...
+You can initialise the SDK like this:
+
+```csharp
+            var flagsmithClient = new FlagsmithClient(Fixtures.ApiKey,
+                httpClient: mockHttpClient.Object,
+                enableClientSideEvaluation: true,
+                cacheConfig: new CacheConfig(true));
+```
+
+If cacheConfig is passed during initialization with its `Enabled` property set to `true`, the SDK will make use cached
```
Comment (dabeeeenster) [inline docs/docs/clients/server-side.md]:
```suggestion
If cacheConfig is passed during initialization with its `Enabled` property set to `true`, the SDK will make use of cached
```

File: docs/docs/clients/server-side.md [hunk 3]
```diff
@@ -1754,6 +1758,25 @@ cache.invalidate(projectLevelCacheKey);
...
+values initially, and then make required API calls as usual and update the cache under the hood.
+
+Currently the TTL for the cache is hard set to 5 minutes.
```
Comment (dabeeeenster) [inline docs/docs/clients/server-side.md]:
```suggestion
The TTL for the cache is 5 minutes, and is not configurable.
```

**Comment by matthewelwell** [comment]
I actually don't think the .NET caching implementation is usable in it's current state since it doesn't implement any method for maintaining the size. This will likely cause a memory leak in any of our customer's applications that utilise the caching and have a large number of identities. See issue [here](https://github.com/Flagsmith/flagsmith-dotnet-client/issues/101).

**Comment by novakzaballa** [comment]
I agree, we also should allow the user to set the TTL.

**Comment by novakzaballa** [comment]
As per the conversation, this docs PR will be closed until the feature is complete.

**Pull request closed without merging**